internal/controllers/machinedeployment/machinedeployment_controller.go

```diff
@@ -399,6 +410,70 @@ func (r *Reconciler) shouldAdopt(md *clusterv1.MachineDeployment) bool {
 	return !util.HasOwner(md.OwnerReferences, clusterv1.GroupVersion.String(), []string{"Cluster"})
 }
 
+func (r *Reconciler) adjustManagedFields(ctx context.Context, obj client.Object) error {
```

Comment:
This will be dropped in favor of the ssa util package that will be introduced in #7921.